### PR TITLE
Reduce mamba prefill allocation overhead

### PR DESCRIPTION
**Pull request by YazhiGao**
## Goal
Reduce CPU overhead in prefill scheduling by batching mamba state slot allocation. The scheduler opens one grouped allocation window for the waiting queue, and normal `mamba_pool.alloc(1)` calls draw from that preallocated batch instead of repeatedly doing per-request allocation bookkeeping.

This batches allocation only. Mamba COW `copy_from` remains immediate while the source cache slot is known to be valid; deferring those copies until group end can race with eviction and read from a freed source slot.

## Changes
- Add `MambaPool.alloc_group_begin()` / `alloc_group_end()` and route single-slot `alloc(1)` through the grouped iterator.
- Wrap waiting-queue prefill scheduling with the mamba pool group API.
- Free any unused preallocated mamba slots when the scheduling pass stops early.

## Audit
- There is no `python/sglang/private` package in OSS, so no private mamba/SWA cache file is touched.
- Current OSS mamba cache implementations already do COW copies immediately; this PR preserves that behavior.

## Testing
- `pre-commit run --files python/sglang/srt/managers/scheduler.py python/sglang/srt/mem_cache/memory_pool.py`
- `git diff --check`
- `python3 -m py_compile python/sglang/srt/managers/scheduler.py python/sglang/srt/mem_cache/memory_pool.py`

---
### CI States

Latest PR Test (Base): :x: [Run #26926278087](https://github.com/sgl-project/sglang/actions/runs/26926278087)
Latest PR Test (Extra): :x: [Run #26926277967](https://github.com/sgl-project/sglang/actions/runs/26926277967)

### CODE REVIEW

File: python/sglang/srt/mem_cache/memory_pool.py [hunk 1]
```diff
@@ -353,7 +353,29 @@ def mamba2_layer_cache(self, layer_id: int):
...
+
+    def alloc_group_end(self):
+        if self._alloc_iter is not None:
+            remaining = list(self._alloc_iter)
+            if remaining:
+                self.free(torch.cat(remaining))
+        self._alloc_iter = None
+
     def alloc(self, need_size: int) -> Optional[torch.Tensor]:
+        if getattr(self, "_alloc_iter", None) is not None and need_size == 1:
```
Comment (ispobock) [inline python/sglang/srt/mem_cache/memory_pool.py]: can we move the `_alloc_iter` attr to the `__init__` with a comment?

**Review by ispobock — APPROVED** [review]
